TeXmacs/progs/init-texmacs-s7.scm

```diff
@@ -529,3 +542,8 @@
                   ;(quit-TeXmacs)
                   ))))))))))))
 
+;; iota is not implemented for s7.
```

Comment:
Do not use tm-define for simple general utility, tm-define has some overhead and is not needed here if you do not plan to leave to the user the possibility to override this functionality dynamically. Look at how the compatibility layer is implemented in (kernel boot compat-s7) and put this function there, as a public define.